File: JUNGOL/Beginner/#2813 소수의 개수/소수의 개수.cpp
#include <iostream>
#include <vector>
using namespace std;

#define MAX_NUM 3000000

int main() {
	// 에라토스테네스의 체
	vector<bool> isPrime(MAX_NUM+1, true);
	isPrime[1] = false;

	for(int i=4; i<=MAX_NUM; i+=2) isPrime[i] = false;
	for(int i=3; i<=MAX_NUM; i+=2){
		if(isPrime[i]){
			for(int j=i*2; j<=MAX_NUM; j+=i)
				isPrime[j] = false;
		}
	}

	int m,n; cin>>m>>n;
	int cnt = 0;
	for(int i=m; i<=n; i++)
		if(isPrime[i]) cnt++;

	printf("%d\n", cnt);
}

File: JUNGOL/Beginner/#2813 소수의 개수/소수의 개수.py
def main():
    MAX_NUM = 3000000

    isPrime = [True for i in range(MAX_NUM+1)]
    isPrime[1] = False

    for i in range(4, MAX_NUM+1, 2): isPrime[i] = False
    for i in range(3, MAX_NUM+1, 2):
        if isPrime[i]:
            for j in range(i*2, MAX_NUM+1, i):
                isPrime[j] = False

    m,n = map(int, input().split())
    cnt = 0
    for i in range(m, n+1):
        if isPrime[i]: cnt += 1

    print(cnt)

main()
